modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalResponse.java

```diff
@@ -85,22 +88,27 @@ public void writeTo(StreamOutput out) throws IOException {
     }
 
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject();
-        builder.field("metric_score", metricScore);
-        builder.startObject("details");
-        for (String key : details.keySet()) {
-            details.get(key).toXContent(builder, params);
-        }
-        builder.endObject();
-        builder.startObject("failures");
-        for (String key : failures.keySet()) {
-            builder.startObject(key);
-            ElasticsearchException.generateFailureXContent(builder, params, failures.get(key), true);
-            builder.endObject();
-        }
-        builder.endObject();
-        builder.endObject();
-        return builder;
+    public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params outerParams) {
+        return Iterators.concat(
+            ChunkedToXContentHelper.singleChunk(
```

Comment:
Another Q: I have seen 3 styles here; have all as single Iterators, have a single chunk, have a single iterator building more stuff:
1. concat of`Iterator.single((b, p) -> b.something())`
2. singleChunk of `(b, p) -> b.something())`
3. `Iterator.single((b, p) -> { b.something(); b.something(); return b; })`

Which one is best, _for small things_? E.g. a bunk of 3/4 simple fields?
I think 3 is the less elegant but more efficient in this case?
Wdyt?

Reply:
I wouldn't expect it to matter, we're consuming these things in a fairly tight loop here:

https://github.com/elastic/elasticsearch/blob/03c9f89bc17c447ce79a0ba239368432996cfd74/server/src/main/java/org/elasticsearch/rest/ChunkedRestResponseBody.java#L111-L116

The only effect of an extra chunk boundary at this level is an extra output size check and a few extra method calls.

Reply:
I thought that was important (the few extra calls); also looking at how concat is implemented, each addition would be an extra `ConcatenatedIterator` object. 
But maybe all gets lost in the overall cost of actually serializing things.

Reply:
> each additional iterator would be an extra `ConcatenatedIterator` object.

That doesn't sound right, you get one `ConcatenatedIterator` for each call to `Iterators#concat`, wrapping potentially many other iterators. Nesting calls to `concat` seems like a bad plan tho.

Reply:
My bad, I probably read it wrong and assumed some head-tail structure when unrolling. Will look at it again (this makes my comment above about the list also bogus)